### ISSUE TITLE
Build fails if enum is defined in a public extension

### ISSUE THREAD

**Issue by donnywdavis** [pointfreeco/swift-case-paths#148]
**Describe the bug**
I have an enum that is defined in an extension that is `public`. The enum inherits the `public` access modifier from the extension, however, the `AllCasePaths` struct and `allCasePaths` properties do not. This causes the build to fail since these two pieces do not have the same access modifier as the enum that is configured with `CasePathable`.

If I add the `public` access modifier directly to the enum then everything works as expected. Even if I move the `public` keyword, we use `SwiftFormat`, that has a rule that moves that modifier to the extension. We would have to disable that rule to make it work. Would it be possible that this could work in both contexts, or is this a limitation of macros?

```swift
// This will fail to compile
public extension ABTest {
    @CasePathable
    enum Identifier {
    }
}

// This will compile without any issues
extension ABTest {
    @CasePathable
    public enum Identifier {
    }
}
```

**Environment**
  - swift-case-paths 1.2.3
  - Xcode 15.1
  - Swift 5.9

**Comment by stephencelis** [comment]
Macros cannot see the parent, so we can't extrapolate the `public` access on the extension, but it does look like we can always apply `public` to these APIs, even if the container is not. Should be fixed when #149 is merged!

**Comment by donnywdavis** [comment]
I figured that it might be a limitation of macros. Thank you for still finding a solution. Really enjoying the possibilities that this package has opened up for us. 😃
